### PR TITLE
fast-bridge: Emit events for each method.

### PR DESCRIPTION
**Pull request by dhruvja**
Emitting events for all instructions so that they can be parsed from the logs and be acted upon.

### CODE REVIEW

**Comment by dhruvja** [comment]
Will `fmt` after review since it results in a huge amount of changes.

File: solana/bridge-escrow/programs/bridge-escrow/src/lib.rs [hunk 1]
```diff
@@ -135,10 +174,22 @@ pub mod bridge_escrow {
...
+        .map_err(|err| {
+            msg!("{}", err);
+            ErrorCode::InvalidEventFormat
+        })?;
+
         Ok(())
     }
 
     // this function is called by Solver
+    #[allow(unused_variables)]
```
Comment (mina86) [inline solana/bridge-escrow/programs/bridge-escrow/src/lib.rs:192]: Why not prefix with underscore?

File: solana/bridge-escrow/programs/bridge-escrow/src/lib.rs [hunk 2]
```diff
@@ -83,16 +99,39 @@ pub mod bridge_escrow {
...
+
+        let current_timestamp = Clock::get()?.unix_timestamp as u64;
+
+        intent.intent_id = intent_id.clone();
         intent.user = user_in;
         intent.token_in = token_in;
         intent.amount_in = amount_in;
-        intent.token_out = token_out;
+        intent.token_out = token_out.clone();
         intent.timeout_timestamp_in_sec = timeout_in_sec;
```
Comment (mina86) [inline solana/bridge-escrow/programs/bridge-escrow/src/lib.rs:110]: Not exactly part of this PR, but is the timeout relative duration or is it a timestamp? This is also what I’ve commented on the other PR.

**Review by mina86 — APPROVED** [review]

File: solana/bridge-escrow/programs/bridge-escrow/src/lib.rs (hunk 2 above)
Comment (dhruvja) [reply solana/bridge-escrow/programs/bridge-escrow/src/lib.rs:110]: resolved in the other PR.

**Pull request merged**